pkg/services/tree/signature.go

```diff
@@ -36,53 +50,66 @@ func (s *Service) verifyClient(req message, cid cidSDK.ID, rawBearer []byte, op
 		return fmt.Errorf("can't get container %s: %w", cid, err)
 	}
 
-	ownerID := cnr.Value.Owner()
+	role, err := roleFromReq(cnr, req)
+	if err != nil {
+		return fmt.Errorf("can't get request role: %w", err)
+	}
 
-	if len(rawBearer) == 0 { // must be signed by the owner
-		// No error is expected because `VerifyDataWithSource` checks the signature.
-		// However, we may use different algorithms in the future, thus this check.
-		pub, err := keys.NewPublicKeyFromBytes(req.GetSignature().GetKey(), elliptic.P256())
-		if err != nil {
-			return fmt.Errorf("invalid public key: %w", err)
-		}
+	basicACL := cnr.Value.BasicACL()
 
-		var actualID user.ID
-		user.IDFromKey(&actualID, (ecdsa.PublicKey)(*pub))
+	if !basicACL.IsOpAllowed(op, role) {
+		return basicACLErr(op)
+	}
 
-		if !actualID.Equals(ownerID) {
-			return errors.New("request must be signed by a container owner")
-		}
+	if !basicACL.Extendable() {
 		return nil
 	}
 
-	var bt bearer.Token
-	if err := bt.Unmarshal(rawBearer); err != nil {
-		return fmt.Errorf("invalid bearer token: %w", err)
-	}
-	if !bearer.ResolveIssuer(bt).Equals(ownerID) {
-		return errors.New("bearer token must be signed by the container owner")
-	}
-	if !bt.AssertContainer(cid) {
-		return errors.New("bearer token is created for another container")
-	}
-	if !bt.VerifySignature() {
-		return errors.New("invalid bearer token signature")
-	}
+	eaclOp := eACLOp(op)
+
+	var tb eacl.Table
+	if len(rawBearer) != 0 && basicACL.AllowedBearerRules(op) {
```

Comment:
I think it might be useful to log with DEBUG level if `len(rawBearer) != 0` and `basicACL.AllowedBearerRules(op)` is false. What do you think?

Reply:
agree, added debug log for that case